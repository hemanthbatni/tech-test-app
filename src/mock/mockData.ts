import { Product } from "../store/productsSlice";

interface ProductsResponse {
  data: {
    search: {
      total: number;
      products: Product[];
    };
  };
}

export const productsMock: ProductsResponse = {
  data: {
    search: {
      total: 4,
      products: [
        {
          id: "product-id-1",
          name: "Prohibition Liquor Company",
          description: "Prohibition Christmas 500ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 200,
            },
            pointsPrice: {
              amount: 25000,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 119,
            },
            pointsPrice: {
              amount: 15000,
            },
          },
        },
        {
          id: "product-id-2",
          name: "Houraisen",
          description: "Houraisen Beshi Tokubetsu Junmai Sake 720ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 134,
            },
            pointsPrice: {
              amount: 23470,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 129.99,
            },
            pointsPrice: {
              amount: 19900,
            },
          },
        },
        {
          id: "product-id-3",
          name: "Royal Salute",
          description:
            "Royal Salute 21 Year Old Blended Scotch Whisky Miami Polo Edition 700ml (Gift Box)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 399,
            },
            pointsPrice: {
              amount: 79800,
            },
          },
        },
        {
          id: "product-id-4",
          name: "Aberlour",
          description:
            "Aberlour 16 Year Old Double Cask Single Malt Scotch Whisky 700ml (Gift Tube)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 199.9,
            },
            pointsPrice: {
              amount: 39980,
            },
          },
        },
        {
          id: "product-id-5",
          name: "Prohibition Liquor Company",
          description: "Prohibition Christmas 500ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 200,
            },
            pointsPrice: {
              amount: 25000,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 119,
            },
            pointsPrice: {
              amount: 15000,
            },
          },
        },
        {
          id: "product-id-6",
          name: "Houraisen",
          description: "Houraisen Beshi Tokubetsu Junmai Sake 720ml",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: "sale",
          wasPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 134,
            },
            pointsPrice: {
              amount: 23470,
            },
          },
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 129.99,
            },
            pointsPrice: {
              amount: 19900,
            },
          },
        },
        {
          id: "product-id-7",
          name: "Royal Salute",
          description:
            "Royal Salute 21 Year Old Blended Scotch Whisky Miami Polo Edition 700ml (Gift Box)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 399,
            },
            pointsPrice: {
              amount: 79800,
            },
          },
        },
        {
          id: "product-id-8",
          name: "Aberlour",
          description:
            "Aberlour 16 Year Old Double Cask Single Malt Scotch Whisky 700ml (Gift Tube)",
          image:
            "https://static.qantasloyalty.com/wine-static-assets/tablet/SPHOURKUUSAKE_1.webp",
          tag: null,
          currentPrice: {
            cashPrice: {
              currencyCode: "AUD",
              amount: 199.9,
            },
            pointsPrice: {
              amount: 39980,
            },
          },
        },
      ],
    },
  },
};
