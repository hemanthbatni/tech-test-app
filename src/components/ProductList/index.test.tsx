import { createEvent, fireEvent, render, screen } from "@testing-library/react";
import { ProductList } from ".";
import { productsMock } from "../../data/mockData";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

jest.spyOn(global.console, "error").mockImplementation();

it("renders component", () => {
  const initialState = { products: [] };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const product = productsMock.data.search.products[0];
  const { container, getByTestId } = render(
    <Provider store={store}>
      <ProductList />
    </Provider>
  );

  expect(container).toMatchSnapshot();
  expect(getByTestId("slider-container")).toBeInTheDocument();
});
