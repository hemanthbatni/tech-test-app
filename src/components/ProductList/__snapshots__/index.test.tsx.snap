// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <div
    class="slider-container"
    data-testid="slider-container"
  >
    <div
      class="slick-slider slick-initialized"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        />
      </div>
    </div>
  </div>
</div>
`;
